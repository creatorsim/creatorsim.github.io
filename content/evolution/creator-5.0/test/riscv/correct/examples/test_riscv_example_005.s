
#
# Creator (https://creatorsim.github.io/creator/)
#

# Sum of the first 10 numbers from 0 to 9
.data
	max: .byte 10

.text
	main: 	    
		la   t0, max
		lb   t0, 0 (t0)
		li   t1, 0
		li   a0, 0
		
	while:	bge  t1, t0, end_while
		add  a0, a0, t1
		addi t1, t1, 1
		beq  zero, zero, while

    end_while: 	li a7, 1
		ecall # print_int

		#return
		jr ra
