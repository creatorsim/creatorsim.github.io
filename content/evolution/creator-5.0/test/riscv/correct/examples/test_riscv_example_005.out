
45
cr[PC]:0xfffffffe; ir[x5,t0]:0xa; ir[x6,t1]:0xa; ir[x10,a0]:0x2d; ir[x17,a7]:0x1; keyboard[0x0]:''; display[0x0]:'45'; 

